Sources/AsyncHTTPClient/AsyncAwait/HTTPClient+execute.swift

```diff
@@ -0,0 +1,166 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if compiler(>=5.5) && canImport(_Concurrency)
+import struct Foundation.URL
+import Logging
+import NIOCore
+import NIOHTTP1
+
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+extension HTTPClient {
+    /// Execute arbitrary HTTP requests.
+    ///
+    /// - Parameters:
+    ///   - request: HTTP request to execute.
+    ///   - deadline: Point in time by which the request must complete.
+    ///   - logger: The logger to use for this request.
+    /// - Returns: The response to the request. Note that the `body` of the response may not yet have been fully received.
+    func execute(
+        _ request: HTTPClientRequest,
+        deadline: NIODeadline,
+        logger: Logger
+    ) async throws -> HTTPClientResponse {
+        try await self.executeAndFollowRedirectsIfNeeded(
+            request,
+            deadline: deadline,
+            logger: logger,
+            redirectState: RedirectState(self.configuration.redirectConfiguration.mode, initialURL: request.url)
+        )
+    }
+}
+
+@available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+extension HTTPClient {
+    private func executeAndFollowRedirectsIfNeeded(
+        _ request: HTTPClientRequest,
+        deadline: NIODeadline,
+        logger: Logger,
+        redirectState: RedirectState?
+    ) async throws -> HTTPClientResponse {
+        let preparedRequest = try HTTPClientRequest.Prepared(request)
+        let response = try await executeWithoutFollowingRedirects(preparedRequest, deadline: deadline, logger: logger)
```

Comment:
This method has nothing really to do with followingRedirects, but cancellation is its feature. WDYT about `executeCancellable`?